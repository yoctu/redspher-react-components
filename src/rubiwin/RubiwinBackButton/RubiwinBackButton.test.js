import React from 'react'
import renderer from 'react-test-renderer'
import { shallow, configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import { RubiwinBackButton } from '../../index'

configure({ adapter: new Adapter() })

/** Snapshots */
it('renders correctly with string', () => {
  const tree = renderer
    .create(<RubiwinBackButton text='test' onClick={() => {}} />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders correctly with element', () => {
  const tree = renderer
    .create(<RubiwinBackButton text={<>test</>} onClick={() => {}} />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})

/** interaction test */
it('should handle click event', () => {
  const MockCallback = jest.fn()
  const wrapper = shallow(
    <RubiwinBackButton text='Rubiwin' onClick={MockCallback} />
  )
  wrapper.simulate('click')
  expect(MockCallback).toHaveBeenCalled()
  wrapper.unmount()
})
