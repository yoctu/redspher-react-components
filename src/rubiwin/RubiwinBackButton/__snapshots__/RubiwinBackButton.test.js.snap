// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with element 1`] = `
<div
  className="MuiBox-root css-1mpub4e"
  onClick={[Function]}
>
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
    data-testid="ArrowBackIosIcon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M11.67 3.87 9.9 2.1 0 12l9.9 9.9 1.77-1.77L3.54 12z"
    />
  </svg>
  test
</div>
`;

exports[`renders correctly with string 1`] = `
<div
  className="MuiBox-root css-1mpub4e"
  onClick={[Function]}
>
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
    data-testid="ArrowBackIosIcon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M11.67 3.87 9.9 2.1 0 12l9.9 9.9 1.77-1.77L3.54 12z"
    />
  </svg>
  test
</div>
`;
